### ISSUE TITLE
How to getHTML after custom node rendering has been applied

### ISSUE THREAD

**Issue by arthurmcgregor** [scrumpy/tiptap#598]
**Is your feature request related to a problem? Please describe.**
We use the tiptap mention/suggestion node to allow users to build custom templates in an edit mode, which are then auto-populated/rendered in the view mode based upon the router/filter applied. For example, users can write templates like `Expenditure is @Expenditure (@Percent of total)` which render as `Expenditure is $15,000 (0.15% of total)`. This works really well and allows users to create written content with live numeric or categorical fields for multiple routes very quickly.

However, my current problem is that I want to extract this text to be used as a page meta description, unique for each route. So what I want is `Expenditure is $15,000 (0.15% of total)` but all I can get is `Expenditure is @Expenditure (@Percent of total)` which would be meaningless in SEO search results!

I have created a (highly simplified) reproduction of our use case here. The actual use case uses vuex and aggregates numeric values & categories within a hierarchy for a dynamic list of fields.

[![Edit shy-glade-4ds9v](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/shy-glade-4ds9v?fontsize=14&hidenavigation=1&theme=dark)

**Describe the solution you'd like**
Some method, similar to getHTML(), except it returns the rendered html rather than the stored html.

**Describe alternatives you've considered**
- I tried creating an Editor instance, and ran both editor.getHTML() and editor.view.dom.innerHTML, but no luck
- I suspect this is probably due to `get view () { return TipTapField }` not being invoked by TipTap/proseMirror, but rather being called by the Vue system instead. In which case, this is probably less of a feature request and more of a question to see if anyone knows how to make this work!

**Additional context**
I realise there are issues with parsing HTML to a string... but I really don't care about whitespace or  html injection risks because those issues I can deal with ;-)

**Comment by vwall** [comment]
@arthurmcgregor TipTap will already do this.  Assuming your TipTapField.vue is setup correctly. FileNode.js is missing the get schema() that defines the attrs and toDOM method that would convert it to the proper HTML.

**Comment by arthurmcgregor** [comment]
So I have found a work-around. Basically, I'm pushing all of the logic into a Vuex store getter function. When given the id, it returns the appropriate string to display. Then I'm using that in both TipTapField.vue and FieldNode.js. To my surprise, editor.getHTML() is reactive and it seems to work pretty well! I updated the codesandbox for anyone else who strikes a similar issue but will close this now as, for me at least, I no longer require this feature.

[![Edit confident-sun-d35vu](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/confident-sun-d35vu?fontsize=14&hidenavigation=1&theme=dark)
